src/Sentry/Protocol/SentryLog.cs

```diff
@@ -0,0 +1,280 @@
+        value = 0L;
+        return false;
+    }
+
+    /// <summary>
+    /// Gets the attribute value associated with the specified key when of type <see cref="double"/>.
+    /// <para>This API is experimental and it may change in the future.</para>
+    /// </summary>
+    /// <remarks>
+    /// Returns <see langword="true"/> if the <see cref="SentryLog"/> contains an attribute with the specified key of type <see cref="double"/>.
+    /// Otherwise <see langword="false"/>.
+    /// </remarks>
+    [Experimental(DiagnosticId.ExperimentalFeature)]
+    public bool TryGetAttribute(string key, out double value)
+    {
+        if (_attributes.TryGetValue(key, out var attribute) && attribute.Type == "double")
+        {
+            value = (double)attribute.Value;
+            return true;
+        }
+
+        value = 0.0;
+        return false;
+    }
+
+    /// <summary>
+    /// Set a key-value pair of <see cref="string"/> data attached to the log.
+    /// <para>This API is experimental and it may change in the future.</para>
+    /// </summary>
+    [Experimental(DiagnosticId.ExperimentalFeature)]
+    public void SetAttribute(string key, string value)
+    {
+        _attributes[key] = new SentryAttribute(value, "string");
+    }
+
+    /// <summary>
+    /// Set a key-value pair of <see cref="bool"/> data attached to the log.
+    /// <para>This API is experimental and it may change in the future.</para>
+    /// </summary>
+    [Experimental(DiagnosticId.ExperimentalFeature)]
+    public void SetAttribute(string key, bool value)
+    {
+        _attributes[key] = new SentryAttribute(value, "boolean");
+    }
+
+    /// <summary>
+    /// Set a key-value pair of <see cref="long"/> data attached to the log.
+    /// <para>This API is experimental and it may change in the future.</para>
+    /// </summary>
+    [Experimental(DiagnosticId.ExperimentalFeature)]
+    public void SetAttribute(string key, long value)
+    {
+        _attributes[key] = new SentryAttribute(value, "integer");
+    }
+
+    /// <summary>
+    /// Set a key-value pair of <see cref="double"/> data attached to the log.
+    /// <para>This API is experimental and it may change in the future.</para>
+    /// </summary>
+    [Experimental(DiagnosticId.ExperimentalFeature)]
+    public void SetAttribute(string key, double value)
+    {
+        _attributes[key] = new SentryAttribute(value, "double");
+    }
+
+    internal void SetAttributes(SentryOptions options)
+    {
+        var environment = options.SettingLocator.GetEnvironment();
+        SetAttribute("sentry.environment", environment);
+
+        var release = options.SettingLocator.GetRelease();
+        if (release is not null)
+        {
+            SetAttribute("sentry.release", release);
+        }
+
+        if (ParentSpanId.HasValue)
+        {
+            SetAttribute("sentry.trace.parent_span_id", ParentSpanId.Value.ToString());
+        }
+
+        SetAttribute("sentry.sdk.name", Constants.SdkName);
```

Comment:
what would we report on `Unity`? 

We have ways to let outer SDKs to override the name, we need that somehow to reflect here.

We might want to improve how we take on the SDK name, today I think it's at event creation time we set the value, then during event building/processing in the SDK we don't override it. And in EventProcessor we set the value if it's null.

Perhaps we want some `internal` property on options that let the value be changed by outer SDKs? not sure, worth checking with @vaind how he did this on Unity/Android/etc

Reply:
Will fix in immediate follow-up when integrating via `Sentry.Extensions.Logging`
See https://github.com/getsentry/sentry-dotnet/issues/4132#issuecomment-2883482296

Reply:
Fixed by reading the SDK-Name and SDK-Version from the Scope via the now available Hub.